### DISCUSSION TITLE
Support for BME688

### DISCUSSION THREAD

**Discussion by muuuh** [arendst/Tasmota#14513] [Ideas]
Hello,

could you please add support for the Bosch BME688 sensor?

I'm using this https://www.adafruit.com/product/5046 sensor. It's reported as i2c 0x77. I'm not sure if the chip-id has changed for the BME688 https://github.com/arendst/Tasmota/blob/dba1f32c8220f21e256634dc6480a970ca71575a/tasmota/xsns_09_bmp.ino#L39

I've found some information here:
https://community.bosch-sensortec.com/t5/Knowledge-base/FAQs-BME688/ta-p/24822

Best regards,

**Comment by Jason2866** [comment]
To early. If you read further, there is no library (yet) available.

**Reply by arendst** [reply]
I think the document is from 2021. I found a library and am using it now but I see no reference to either BME680 or BME688 so it seems it does not show the difference.

It add 2k5 to the code footprint.

Will release soon.

**Comment by arendst** [comment]
Try latest development branch

**Comment by muuuh** [comment]
Hello,

thank you very much! It worked!

I compiled from the dev branch via https://gitpod.io/#https://github.com/benzino77/tasmocompiler with the flags
```
#define USE_I2C
#define USE_BMP
#define USE_BME680
#define USE_BME68X
```

and `i2cscan` and `i2cdriver` showed the driver 10 (somehow didn't without less flags) and the webserver displayed all relevant values (redacted below).

```
BME680 Temperature	… °C
BME680 Humidity		… %
BME680 Dew point	… °C
BME680 Pressure		… hPa
BME680 Gas		… kΩ
```

Thanks again!

Best regards,

**Comment by alexwhittemore** [comment]
Just wanna say thanks from me as well!

**Comment by Scentyoo** [comment]
Hello, I used the Tasmota firmware and I was able to get the data directly: http://ota.tasmota.com/tasmota/release/tasmota-DE.bin
I used: Adafruit BME688 with makershop Wemos D1 Mini (with Tasmota 11.0.0 (sensors))

In which case should I follow your instructions? What is the diffence to my approach?

**Reply by sfromis** [reply]
Not sure what you really ask about. The sensors binary includes BME68x support, the non-sensors tasmota-DE does not. If you want german language and BME68x support at the same time, you'll have to build a custom binary.

**Reply by Scentyoo** [reply]
Ok, I just realized I probably installed the sensors binary (and not the tasmota-DE) which supports the BME688 out of the box, correct? Compiling like in this post is normally not necessary anymore.

**Reply by sfromis** [reply]
A custom build is indeed only required if the precompiled tasmota-sensors (including BME68x sensor support) is not "enough" for you.

**Reply by Scentyoo** [reply]
Ok, perfect! Thanks!

**Comment by Scentyoo** [comment]
Now I have another question when working with the sensor. As I understand, I only get one value for "gas" which is an overall VOC content. In this Bosch tutorial it is explained that there is a way of recognizing coffee: https://www.bosch-sensortec.com/software-tools/software/BME688-software/
Is there any way to reproduce this (record heater profiles, generate a BSEC config file and use this file to detect coffee). Or is there at least any way to detect the air quality?

**Reply by sfromis** [reply]
At the physical level, the sensor can absolutely only give a resistance number for the tin oxide sensor. Nothing more. Can't smell anything specific.

The rest, including undefined "air quality", is a speculative "voodoo" software package, and not part of Tasmota.

**Reply by corbolais** [reply]
Just found out about the BME688 and this coffee thing: https://www.bosch-sensortec.com/software/bme/docs/process-steps/collect-specimens.html

Maybe the AI based adaptation can be imported at compile time?

Edit: the coffee aka beverages recognition thing is based on the evaluation board featuring eight (8) BME688 sensors. This way, it can plausibly detect/discern VOC mixtures to some degree.
